python/rmm/_lib/device_buffer.pyx

```diff
@@ -34,15 +37,16 @@ cdef class DeviceBuffer:
     def __cinit__(self, *,
                   uintptr_t ptr=0,
                   size_t size=0,
-                  uintptr_t stream=0):
+                  CudaStreamView stream=CudaStreamView()):
```

Comment:
I believe using `CudaStreamView()` in a function signature is problematic as it will run on module import. This means that the CUDA driver and/or CUDA Context will be initialized and we've historically made the libraries able to be imported without a GPU and without initializing the driver / cuda context.

I think this is the right approach as opposed to using `None`, so maybe we just need to make a lazily initialized object? Where instead of using `stream.c_obj.get()` everywhere we could have a helper `get_raw_ptr` function and in the lazy object the `get_raw_ptr` function can initialize the object under the hood.

Alternatively, we could make all `CudaStreamView` objects lazily initialized.

Reply:
Actually, in hindsight this should be okay because the default constructor returns `cudaStreamDefault` which is just a `#define` as opposed to calling into a CUDA API. Resolving this.

Reply:
Correct.